### ISSUE TITLE
Usage of typescript-nx-imports-plugin outside of Nx Console (Neovim)

### ISSUE THREAD

**Issue by AkisArou** [Cammisuli/monodon#18]
Any ideas how can it be done? I've seen that you are using nvim and maybe you could figure it out! I tried with no luck! (newbie to nvim). Thanks!

**Comment by Cammisuli** [comment]
Hmm, we’d probably need to make some kind of plugin that configures the typescript language server. 

Then the input for this need to be all the projects in the workspace with the full path to their tsconfig file.

**Comment by AkisArou** [comment]
Yeah I think this is the way.
Yesterday I implemented a dirty coc.nvim plugin.
I checked the nx-console code and saw that the interface that coc exposes, is like VSCode's.
I got it working!
You can check it out at [https://github.com/AkisArou/coc-nx](url), and also read README for instructions.
I also have [a repository](https://github.com/AkisArou/nx-nvim) that I am experimenting with nx & neovim, if you want to try out the coc implementation.
You can check the tsserver logs with :CocCommand tsserver.openTsServerLog, and the plugin logs with :CocOpenLog.

But I think the nvim lsp is the way to go, and not coc.
So if you are familiar with nvim lsp, we can try it out! I already started experimenting!

**Comment by Cammisuli** [comment]
Oh awesome! I wont have too much time to experiment with the nvim lsp, but I know it's possible 😄

**Comment by AkisArou** [comment]
Oh ok! I will try to find time to implement it and I'll keep in touch!

**Comment by Cammisuli** [comment]
Yup, thats exactly it @antoineviscardi. As long as you can configure the tsserver lsp to include plugins (which need to be installed in node_modules or somewhere where the language server can load it from), and then configure said plugin, it should work. 

Here's where we do the configuration of the imports plugin in Nx Console: https://github.com/nrwl/nx-console/blob/f5ba74fc234d1787ad4f9d269d804b03eb6c0e45/libs/vscode/typescript-plugin/src/lib/typescript-plugin.ts#L108-L113

And here's where we configure the tsserver in VSCode to include the imports plugin:
https://github.com/nrwl/nx-console/blob/f5ba74fc234d1787ad4f9d269d804b03eb6c0e45/apps/vscode/package.json#L58-L63

**Comment by faileon** [comment]
Hi folks, did anyone manage to make any progress with this? @antoineviscardi ? I've recently switched to neovim and I love it, but pretty much all my projects are managed with NX and `Go To Reference` is not working right with nvim-lsp and it's driving me crazy :]. I could give this a try but I also have no experience with tsserver plugins and LSPs.

**Comment by faileon** [comment]
@Ajaymamtora check out https://github.com/pmizio/typescript-tools.nvim/issues/124
